manifests/resource/upstream.pp

```diff
@@ -49,6 +50,7 @@
   $upstream_fail_timeout                    = '10s',
   $upstream_max_fails                       = undef,
   Enum['http', 'stream'] $upstream_context  = 'http',
+  $upstream_weight                          = undef,
```

Comment:
Could you add the data type? if its integer:  `Optional[Integer] $upstream_weight = undef`